File: core/model/parameter.go
package model

type Parameter struct {
	BaseModel
	ParamType     *int   `gorm:"column:param_type" json:"paramType,omitempty"`
	ParamKey      string `gorm:"column:param_key;size:40" json:"paramKey"`
	ParamName     string `gorm:"column:param_name;size:40" json:"paramName"`
	ParamPosition string `gorm:"column:param_position;size:20" json:"paramPosition,omitempty"`
	ParamDesc     string `gorm:"column:param_desc;size:200" json:"paramDesc,omitempty"`
	DataType      string `gorm:"column:data_type;type:text" json:"dataType,omitempty"`
	Required      *int   `gorm:"column:required" json:"required,omitempty"`
	SourceType    string `gorm:"column:source_type;size:8" json:"sourceType,omitempty"`
	SourceID      *int64 `gorm:"column:source_id" json:"sourceId,omitempty"`
}

func (Parameter) TableName() string { return "t_parameter" }

File: server/src/main/java/com/javaoffers/nexus/core/model/Parameter.java
package com.javaoffers.nexus.core.model;

import com.javaoffers.brief.modelhelper.anno.BaseModel;
import com.javaoffers.brief.modelhelper.anno.BaseUnique;
import com.javaoffers.brief.modelhelper.anno.ColName;
import com.javaoffers.brief.modelhelper.anno.derive.flag.IsDel;
import lombok.Data;

import java.util.Date;

@BaseModel("t_parameter")
@Data
public class Parameter {
    @BaseUnique
    private Long id;
    private IsDel deleted;
    private Date createdAt;
    private Long createdBy;
    private Date updatedAt;
    private Long updatedBy;

    @ColName(value = "id", excludeColAll = true)
    private Long countId;

    private Integer paramType;
    private String paramKey;
    private String paramName;
    private String paramPosition;
    private String paramDesc;
    private String dataType;
    private Integer required;
    private String sourceType;
    private Long sourceId;
}
